activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb

```diff
@@ -1510,6 +1530,25 @@ def valid_primary_key_options # :nodoc:
       end
 
       private
+        MAX_GENERATED_INDEX_NAME_BYTES = 62
+
+        def generate_index_name(table_name, column)
+          name = "index_#{table_name}_on_#{Array(column) * '_and_'}"
+          return name if name.bytesize <= MAX_GENERATED_INDEX_NAME_BYTES
+
+          # Remove _and_ between columns
+          name = "index_#{table_name}_on_#{Array(column) * '_'}"
+          return name if name.bytesize <= MAX_GENERATED_INDEX_NAME_BYTES
+
+          # Remove table_name, add hash for uniqueness
+          hashed_identifier = OpenSSL::Digest::SHA256.hexdigest(name).first(10)
+
+          name = "index_on_#{Array(column) * '_'}"
```

Comment:
```suggestion
          name = ""index_#{table_name}_on_#{Array(column) * '_'}"
```

If we're gonna truncate anyway, I'd say the table name is the part that is the more relevant to try to include?

Reply:
IMO in most interesting contexts where you see indexes mentioned, they're inherently table-scoped: if you're looking at which index is being used to scan the `users` table (or writing a hint to force it), the part that distinguishes this index from its peers is which columns it covers.

Reply:
Yeah, my opinion is that we have exactly two patterns: the "there's plenty of room" historical default that includes table + columns, and the "rest of the time" fallback that only lists columns plus a digest (protecting against inter-table column collisions, plus the edge case of collisions caused by truncation).

(I'm also still tempted to give it a different shorter prefix like `ix_`, both to emphasize that this is a Mode Two name, and to conserve name chars given they're apparently not plentiful... also not strong on that though.)

Reply:
Thanks! I pushed an update based on this convo. I like the idea of keeping it simple with only 2 versions. 

Here's how it works now:
If the original version is too long, we immediately fallback to the "short version"

short format example:
```
ix_on_foo_bar_first_name_last_name_administrator_5939248142
```